contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -870,8 +946,11 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
         );
 
         // If there are some queue elements pending:
-        if (getQueueLength() - _nextQueueIndex > 0) {
-            Lib_OVMCodec.QueueElement memory nextQueueElement = getQueueElement(_nextQueueIndex);
+        if (_queueLength - _nextQueueIndex > 0) {
```

Comment:
```suggestion
        if (_nextContext.numSubsequentQueueElements > 0 && _queueLength - _nextQueueIndex > 0) {
```
Am I right that we can further optimize this way?  From stats below it looks like this isn't the biggest source of cost ATM, but if it sounds right to you, seems worth doing because it's this easy.

Reply:
would also update the comment a line above accordingly

Reply:
This is actually quite a large source of cost, so I do think this is worthwhile. Although are we sure that this is valid? Do we only need to do this check when there are subsequent queue elements?

Reply:
@ben-chain can you take another look at this?

Reply:
Just did another playthrough; it looks like this is insufficient in the case that *all* subsequent contexts have no queue elements.

@smartcontracts If you think there are big savings, then comparing the `_finalContext` to the next pending queue element in `_validateFinalBatchContext` would make it work.  This could be done conditionally; only if the final batch context has no subsequent queue elements.

Reply:
Eh okay this seems sufficiently complex that we can save it for a future set of optimizations. This PR gets our gas costs down pretty hard already.

Reply:
I'll make a note of this in my scratch pad.